src/utilities/units.ts

```diff
@@ -36,7 +36,7 @@ export function toExponential(val: number, decimals: number = 0): string {
 
 // According to MDN, toFixed only works for up to 20 decimals
 export function toFixed(val: number, decimals: number = 0): string {
-    if (isFinite(val) && isFinite(decimals) && decimals >= 0 && decimals <= 20) {
+    if (val && isFinite(val) && isFinite(decimals) && decimals >= 0 && decimals <= 20) {
```

Comment:
`isFinite(val)` only (without `val &&`) may be sufficient for checking `val`. When `val` is zero, `val &&` would cause `toFixed` to return `String(val)`, which is an unexpected behavior.

Reply:
I am worried it may cause a hard crash since isFinite(null) = true.

Reply:
@crocka I see. I suggest we replace `null` with `NaN` in `DistanceMeasuringStore` to prevent such crash. Also change `x != null` and `y != null` to `isFinite(x)` and `isFinite(y)`.

Reply:
Yup, please review the latest commit.

Reply:
Yup, please check the lastest commit.